lib/puppet/provider/exec/powershell.rb

```diff
@@ -78,8 +78,17 @@ def execute_resource(powershell_code, resource)
     end
     timeout_ms = resource[:timeout].nil? ? nil : resource[:timeout] * 1000
     environment_variables = resource[:environment].nil? ? [] : resource[:environment]
+    
+    if powershell_code.is_a?(Array)
```

Comment:
You are on the right track here but I think this needs some more work.

I wonder if we could add some shell sanitisation to the strings passed through.

For example, if the command is an array, we could safely join the string together and if it is a string we could ensure that it is safely escaped.

There are some methods in the Shellwords library that we could make use of. However, I'm not entirely sure how they would impact commands sent to PowerShell.

Lets pair next week and explore some options together.